boards: arm: add nrf9131ek_nrf9131

maxd-nordic opened the pull request:
This patch adds the nRF9131-EK board.

maxd-nordic committed:
boards: arm: add nrf9131ek_nrf9131

This patch adds the nRF9131-EK board.

Signed-off-by: Maximilian Deubel <maximilian.deubel@nordicsemi.no>

nordicjm commented on boards/arm/nrf9131ek_nrf9131/nrf9131ek_nrf9131_common.dtsi:
```diff
@@ -0,0 +1,227 @@
+		slot1_ns_partition: partition@c0000 {
+			label = "image-1-nonsecure";
+		};
+		scratch_partition: partition@f0000 {
+			label = "image-scratch";
+			reg = <0x000f0000 0xa000>;
+		};
```
really should not be using scratch in 2023

maxd-nordic replied:
fixed

nordicjm commented on boards/arm/nrf9131ek_nrf9131/nrf9131ek_nrf9131.yaml:
```diff
@@ -0,0 +1,21 @@
+  - gnuarmemb
+  - xtools
+  - zephyr
+ram: 88
```
88KB of RAM?

maxd-nordic replied:
should I put 128kB of usable ram (for non-secure) or 256kB of total SoC ram?

nordicjm replied:
256 probably since tests can use the full 256KiB (I presume?)

maxd-nordic commented:
@nordicjm these things were copied from nrf9161dk. should they be fixed there as well?

maxd-nordic commented:
For non-secure builds, this one might be needed: https://github.com/zephyrproject-rtos/trusted-firmware-m/pull/97

nordicjm commented:
> @nordicjm these things were copied from nrf9161dk. should they be fixed there as well?

Yes

nordicjm commented on boards/arm/nrf9161dk_nrf9161/nrf9161dk_nrf9161.yaml:
```diff
@@ -6,7 +6,7 @@ toolchain:
   - zephyr
-ram: 88
+ram: 256
 flash: 1024
```
Scratch partition should be removed from 9161 also (can do it in a different PR)

maxd-nordic replied:
just added the change for the nrf9160dk as well :+1:

joerchan reviewed (dismissed):
TF-M Integration looks good, ran the integration tests and they are passing.

maxd-nordic committed:
boards: arm: nrf9161dk: remove scratch partition

This patch removes the now unused scratch partition and enlarges
the application slots instead.

Signed-off-by: Maximilian Deubel <maximilian.deubel@nordicsemi.no>

nordicjm commented:
```
		slot1_ns_partition: partition@c5000 {
			reg = <0x000c0000 0x35000>;
```
These 2 conflict

maxd-nordic commented:
> ```
> 		slot1_ns_partition: partition@c5000 {
> 			reg = <0x000c0000 0x35000>;
> ```
> 
> These 2 conflict

Sorry! Clumsy editing here. Maybe these partition configs should be in one file?

carlescufi commented:
@anangl can you take a look?

anangl commented on boards/arm/nrf9131ek_nrf9131/nrf9131ek_nrf9131_common.dtsi:
```diff
@@ -0,0 +1,225 @@
+	pinctrl-names = "default", "sleep";
+};
+
+arduino_serial: &uart1 {
```
Does this board actually contain the Arduino header (there is no related nexus node definition in this file)?

maxd-nordic replied:
It doesn't. I'll remove the arduino labels.

anangl commented on boards/arm/nrf9131ek_nrf9131/nrf9131ek_nrf9131_common.dtsi:
```diff
@@ -156,6 +156,9 @@ arduino_spi: &spi3 {
 		reg = <0>;
 		spi-max-frequency = <8000000>;
 		size = <268435456>;
+		has-dpd;
+		t-enter-dpd = <3000>;
+		t-exit-dpd = <40000>;
```
This should be moved to the first commit ("boards: arm: add nrf9131ek_nrf9131").

anangl commented on boards/arm/nrf9131ek_nrf9131/nrf9131ek_nrf9131_ns.yaml:
```diff
@@ -0,0 +1,18 @@
+  - xtools
+  - zephyr
+ram: 128
+flash: 276
```
Is this correct? `slot0_ns_partition` size is 212 kB (0x35000).

anangl commented on boards/arm/nrf9161dk_nrf9161/nrf9161dk_nrf9161.yaml:
```diff
@@ -6,7 +6,7 @@ toolchain:
   - xtools
   - zephyr
-ram: 88
+ram: 256
```
I believe this should not be changed. `zephyr,sram` points to `sram0_s` and the size of this node is `DT_SIZE_K(88)`, so that's the RAM size available for samples and tests.

joerchan commented:
@carlescufi I'd like https://github.com/zephyrproject-rtos/zephyr/pull/63515 to go in first. That way we don't have to modify TF-M with the changes in https://github.com/zephyrproject-rtos/trusted-firmware-m/pull/97 in order to get the NS board working.

joerchan commented:
@maxd-nordic Did you rebase this PR and check that the NS board works?

maxd-nordic commented:
> @maxd-nordic Did you rebase this PR and check that the NS board works?

Just checked - all tests pass! :)

mbolivar-ampere reviewed (commented):
ACK on the tests/lib/devicetree/devices/testcase.yaml changes

nordicjm commented on boards/arm/nrf9131ek_nrf9131/nrf9131ek_nrf9131_ns_defconfig:
```diff
@@ -0,0 +1,35 @@
+
+CONFIG_PINCTRL=y
+
+# enable PMIC
+CONFIG_I2C=y
+CONFIG_REGULATOR=y
+CONFIG_SENSOR=y
+CONFIG_NPM1300_CHARGER=y
```
Why are these needed to be on by default for _ns builds?

maxd-nordic replied:
this is because of the charger function - it has to be configured for the hardware to work

maxd-nordic replied:
Because the board will brown out without that configuration. We cannot safely run firmware without the PMIC configured.

anangl commented on boards/arm/nrf9131ek_nrf9131/nrf9131ek_nrf9131.yaml:
```diff
@@ -0,0 +1,17 @@
+  - gnuarmemb
+  - xtools
+  - zephyr
+ram: 256
```
This should match the size of the node chosen as `zephyr,sram`, which is `sram0_s` in this case, so it should be 88 kB.

maxd-nordic replied:
done

anangl commented on boards/arm/nrf9131ek_nrf9131/Kconfig.defconfig:
```diff
@@ -0,0 +1,53 @@
+config BOARD
+	default "nrf9131ek_nrf9131"
+
+# By default, if we build for a Non-Secure version of the board,
+# enable building with TF-M as the Secure Execution Environment.
+config BUILD_WITH_TFM
+	default y if BOARD_NRF9131EK_NRF9131_NS
+
+if BUILD_WITH_TFM
+
+# By default, if we build with TF-M, instruct build system to
+# flash the combined TF-M (Secure) & Zephyr (Non Secure) image
+config TFM_FLASH_MERGED_BINARY
+	bool
+	default y
+
+endif # BUILD_WITH_TFM
+
```
This should be removed after #63515 is merged.

maxd-nordic replied:
done

anangl commented on boards/arm/nrf9161dk_nrf9161/nrf9161dk_nrf9161_common.dtsi:
```diff
@@ -230,16 +230,12 @@ arduino_spi: &spi3 {
 		};
-		slot1_ns_partition: partition@c0000 {
+		slot1_ns_partition: partition@c5000 {
 			label = "image-1-nonsecure";
 		};
-		scratch_partition: partition@f0000 {
-			label = "image-scratch";
-			reg = <0x000f0000 0xa000>;
-		};
 		storage_partition: partition@fa000 {
 			label = "storage";
 			reg = <0x000fa000 0x00006000>;
```
Shouldn't this be done like in dc431a7397a629c9fb7e3a27a803d11bf1103389? @joerchan

maxd-nordic replied:
I suppose this depends on how you want to use the freed up space - I put it in the app image.

joerchan replied:
As long as the TF-M partitions are placed correctly, they don't follow the configuration you put here, they are statically defined.

maxd-nordic replied:
Is this documented somewhere? It seems a bit hard to understand which partition config ends up being used if parts of the configuration are overwritten by build system magic.

joerchan replied:
No. It's not overwritten, it's ignored ;) However in NCS the partition manager symbols are used. So this is only a zephyr restriction.

anangl commented on boards/arm/nrf9131ek_nrf9131/board.cmake:
```diff
@@ -0,0 +1,14 @@
+  set_property(TARGET runners_yaml_props_target PROPERTY hex_file tfm_merged.hex)
+endif()
+
+# TODO: change to nRF9131_xxAA when such device is available in JLink
+board_runner_args(jlink "--device=nRF9160_xxAA" "--speed=4000")
```
It looks like support for nRF9131 and nRF9161 was added in [v7.92m](https://www.segger.com/downloads/jlink/ReleaseNotes_JLink.html).

kartben commented:
Hey @carlescufi @maxd-nordic, this board has no README... Please add at your earliest convenience.

kartben commented:
I realized that the board doesn't seem to be publicly available yet. While that's not necessarily an issue per se, it still must have documentation in the form of a README in the tree (see https://docs.zephyrproject.org/latest/hardware/porting/board_porting.html#contributing-your-board)

kartben commented:
> Hey @anangl @maxd-nordic @carlescufi, this board has no README... Please add at your earliest convenience.

Ping - just want to make sure you guys saw this. Thanks!

maxd-nordic commented:
> > Hey @anangl @maxd-nordic @carlescufi, this board has no README... Please add at your earliest convenience.
> 
> Ping - just want to make sure you guys saw this. Thanks!

Yes, I got your request.
